pkg/cli/admin/catalog/mirrorer.go

```diff
@@ -129,11 +129,21 @@ func imagesFromDb(file string) (map[string]struct{}, error) {
 }
 
 func mappingForImages(images map[string]struct{}, src, dest imagesource.TypedImageReference, maxComponents int) (mapping map[imagesource.TypedImageReference]imagesource.TypedImageReference, errs []error) {
-	// don't do any name mangling when not mirroring to a real registry
-	// this allows us to assume the names are preserved when doing multi-hop mirrors that use a file or s3 as an
-	// intermediate step
 	if dest.Type != imagesource.DestinationRegistry {
```

Comment:
Thanks for taking this on after pairing on it!

Only thing I'm wondering is since we have `dest.Type != imagesource.DestinationRegistry` multiple times if there is a way to abstract this out by having the dest type fulfill an interface that includes a `mappingForImages` function so it's a bit more abstract. Not really worth it at this point, but maybe in the future if the mapping starts to drift more between the different destinations. 

/lgtm